### ISSUE TITLE
UnusedPrivateElements False positive

### ISSUE THREAD

**Issue by bendavies** [slevomat/coding-standard#1045]
Hi,

Here's a `UnusedPrivateElements` false positive:

```php
<?php

declare(strict_types=1);

namespace App;

use Generator;

class Test
{
    public static function foo(): Generator
    {
        yield static fn (self $test): void => $test->iAmUsed();
    }

    private function iAmUsed(): void
    {
    }
}
```

```sh
vendor/bin/phpcs --sniffs=SlevomatCodingStandard.Classes.UnusedPrivateElements Test.php
E 1 / 1 (100%)



FILE: Test.php
----------------------------------------------------------------------
FOUND 1 ERROR AFFECTING 1 LINE
----------------------------------------------------------------------
 16 | ERROR | Class Test contains unused private method iAmUsed().
----------------------------------------------------------------------
```

Thanks!

**Comment by bendavies** [comment]
hey @kukulich, any idea is this is easily resolvable? i'm taking a look but not being familiar with the code is causing me trouble!

**Comment by ondrejmirtes** [comment]
@bendavies I have some news for you :) https://phpstan.org/blog/detecting-unused-private-properties-methods-constants

TL;DR @kukulich decided to deprecate UnusedPrivateElementsSniff in Slevomat Coding Standard, PHPStan should be used for this purpose instead. Make sure you're on 0.12.33 (the latest release) and enable bleedingEdge.neon in your phpstan.neon:

```neon
includes:
	- vendor/phpstan/phpstan/conf/bleedingEdge.neon
```

**Comment by kukulich** [comment]
Marked as deprecated: https://github.com/slevomat/coding-standard/commit/655524a1a70b6b1cc0f75bb5c1b7c6b8286dad86

**Comment by bendavies** [comment]
nice, thanks!
